.github/workflows/pylint.yml

```diff
@@ -0,0 +1,45 @@
+
+
+name: Pylint
+
+on:
+  workflow_dispatch:
+  push:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8  # maybe better not hardcoded
+      - name: Install and configure poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+      - name: Disable modern installer
+        run: poetry config installer.modern-installation false
```

Comment:
Do you really need to disable the modern installer? Most packages are working fine with it, and it speeds up installation.

Have a try without these lines (or even just try `poetry install` locally, after `poetry env remove python`, with a recent enough version of Poetry, i.e >= 1.4.1 and with this setting set to `true`, or just unset).